src/installJava.ts

```diff
@@ -54,6 +56,11 @@ export function installJava({
         .then(() => outputChannel.appendLine(`${java} installed`))
         .then(() => pcp.exec(`${jabbaPath} which --home ${java}`))
         .then((e: pcp.Output) => outputToString(e.stdout).trim());
+    })
+    .catch((err: Error) => {
+      console.debug(err);
+      outputChannel.appendLine(err.message);
+      return undefined;
```

Comment:
I would probably rethrow the error here, so that the clients can handle the error by using `.catch` themselves.

```suggestion
      throw err;
```

Reply:
Thanks! I was thinking we should be able to do it easier.

Reply:
Yeah, one day I'll refactor this using `TaskEither` (which is essentially bi-functor `IO` in `fp-ts`) so that we can use more familiar and powerful combinators to manage it. For the time being this is 👍